src/software/proto/primitive/primitive_msg_factory.h

```diff
@@ -116,4 +136,17 @@ std::unique_ptr<TbotsProto::PrimitiveNew> createAutokickMovePrimitive(
  *
  * @return Pointer to Stop Primitive Message
  */
-std::unique_ptr<TbotsProto::PrimitiveNew> createStopPrimitive(StopType stop_type);
+std::unique_ptr<TbotsProto::Primitive> createStopPrimitive(bool coast);
+
+/**
+ * Creates a new DirectControl Primitive AI could output this primitive to control the
+ * linear velocity, angular velocity, and dribbler speed of a specific robot
+ *
+ * @param velocity x/y velocity vector
+ * @param angular_velocity The angular velocity
+ * @param dribbler_rpm The dribbler speed in rpm
+ *
+ * @return Pointer to the DirectControl Primitive
+ */
+std::unique_ptr<TbotsProto::Primitive> createDirectVelocityPrimitive(
```

Comment:
```suggestion
std::unique_ptr<TbotsProto::Primitive> createDirectControlPrimitive(
```
seems like this is also missing fields such as charge_mode, chick_command and the option of using per_wheel_control

Reply:
hm I see how this can be confusing. This is really just used for the handheld controller generator (i.e. Xbox), so I can just move it over there